workflow/workflow.go

```diff
@@ -55,6 +55,12 @@ func (e *executableWorkflow) DAG() dgraph.DirectedGraph[*DAGItem] {
 	return e.dag
 }
 
+// Namespaces returns a namespaced collection of objects for the inputs
+// and outputs of each stage in the step's lifecycles.
```

Comment:
This seems to get closer to the point than the others, but, we're describing the `Namespaces` through a "namespaced collection of objects"?? That almost sounds circular.

Mostly, I can see a rough implication here around "namespaces" and "inputs"/"outputs" from which I might infer some context around the tiered map, but there's nothing explicit about what I'd expect to find at the first level or the second level.

I'm also getting the feeling that this oft-used `map[string]map[string]*schema.ObjectSchema` might benefit from being a named (and documented) type.